bi/pkg/cluster/pulumi.go

```diff
@@ -23,54 +24,49 @@ const (
 
 type pulumiProvider struct {
 	initSuccessful bool
+	spec           *specs.InstallSpec
 
 	cfg auto.ConfigMap
 	projectName,
-	slug,
 	workDirRoot string
 
 	pulumiHome auto.LocalWorkspaceOption
 	pulumi     auto.LocalWorkspaceOption
 	envVars    auto.LocalWorkspaceOption
 }
 
-func NewPulumiProvider(slug string) Provider {
+func NewPulumiProvider(spec *specs.InstallSpec) Provider {
```

Comment:
Passing the full spec allows us to start infering information from the spec instead of hardcoding.